src/main/java/ui/services/CycleTimeService.java

```diff
@@ -42,51 +61,61 @@ public ObservableList<XYChart.Data<String, Number>> getStories() {
-        LocalDate end = DateUtil.toLocal(sprint.getEstimatedFinish());
+    private List<CycleTimeEntry<taiga.model.query.tasks.Task>> getAllTaskCycleTime() {
+        List<CycleTimeEntry<taiga.model.query.tasks.Task>> cycleTimes = rawTasks.parallelStream().map(TaskUtils::getCycleTimeForTask).toList();
+        LocalDate start = DateUtil.toLocal(startDate);
+        LocalDate end = DateUtil.toLocal(endDate);
         List<LocalDate> dates = start.datesUntil(end.plusDays(1)).toList();
-        List<CycleTimeEntry> finalCycleTimes = new ArrayList<>(cycleTimes);
+        List<CycleTimeEntry<taiga.model.query.tasks.Task>> finalCycleTimes = new ArrayList<>(cycleTimes);
 
         for (LocalDate date : dates) {
-            finalCycleTimes.add(new CycleTimeEntry(DateUtil.toDate(date), null, false));
+            CycleTimeEntry<taiga.model.query.tasks.Task> entry = new CycleTimeEntry<>(null, DateUtil.toDate(date), null, false);
+            entry.setTooltipCallback(new CycleTimeTaskTooltip());
+            finalCycleTimes.add(entry);
         }
 
         return finalCycleTimes;
     }
 
-    private List<CycleTimeEntry> getAllUserStoryCycleTime() {
-        List<CycleTimeEntry> cycleTimes = sprint.getUserStories().parallelStream().map(UserStoryUtils::getCycleTimeForUserStory).toList();
-        LocalDate start = DateUtil.toLocal(sprint.getEstimatedStart());
-        LocalDate end = DateUtil.toLocal(sprint.getEstimatedFinish());
+    private List<CycleTimeEntry<UserStoryInterface>> getAllUserStoryCycleTime() {
+        List<CycleTimeEntry<UserStoryInterface>> cycleTimes = rawUserStories.parallelStream().map(UserStoryUtils::getCycleTimeForUserStory).toList();
+        LocalDate start = DateUtil.toLocal(startDate);
+        LocalDate end = DateUtil.toLocal(endDate);
         List<LocalDate> dates = start.datesUntil(end.plusDays(1)).toList();
-        List<CycleTimeEntry> finalCycleTimes = new ArrayList<>(cycleTimes);
+        List<CycleTimeEntry<UserStoryInterface>> finalCycleTimes = new ArrayList<>(cycleTimes);
 
         for (LocalDate date : dates) {
-            finalCycleTimes.add(new CycleTimeEntry(DateUtil.toDate(date), null, false));
+            CycleTimeEntry<UserStoryInterface> entry = new CycleTimeEntry<>(null, DateUtil.toDate(date), null, false);
+            entry.setTooltipCallback(new CycleTimeUserStoryTooltip());
+            finalCycleTimes.add(entry);
         }
 
         return finalCycleTimes;
     }
 
-    private void updateCycleTimes(ObservableList<XYChart.Data<String, Number>> data, List<CycleTimeEntry> entries) {
+    private <T> void updateCycleTimes(ObservableList<XYChart.Data<String, Number>> data, List<CycleTimeEntry<T>> entries) {
         SimpleDateFormat format = new SimpleDateFormat("MMM dd");
+        List<CycleTimeEntry<T>> sortedEntries = entries.stream()
+                .filter(t -> t.getStartDate() != null)
+                .filter(t -> t.getStartDate().after(startDate) && t.getStartDate().before(endDate))
```

Comment:
Should be resolved by https://github.com/cgjeffries/SER516-Team-Boston/pull/37/commits/a3c4221e85c4752aaa5a0ee1fdf6610cd8bf54b8. Ran into some issues where dates were out of order, and while I think I just simply forgot to save my code, others should verify the following:
1. Date range is correct
2. Dates are ordered correctly

Reply:
Research indicates a history of several graphical bugs related to data ordering for charts. We have traced down the state of the points in the debugger all the way down to the XYChart.Data array that the chart receives, and everything remains sorted, yet in the UI, it is unsorted. We will note this and may address it in the future, but for the sake of time, this will be acknowledged as a JavaFX bug.